frameworklauncher/src/main/java/com/microsoft/frameworklauncher/common/utils/ValueRangeUtils.java

```diff
@@ -355,7 +386,7 @@ public static String toEnviromentVariableString(List<ValueRange> portRanges, Map
           portsString.append(";");
         } else {
           //if user not specified ports, assign the allocated ContainerPorts to each port label.
-          List<ValueRange> assignPorts = ValueRangeUtils.getSubRange(portRanges, ports.getCount(), basePort);
+          List<ValueRange> assignPorts = ValueRangeUtils.getSubRangeSequence(portRanges, ports.getCount(), basePort);
```

Comment:
>getSubRangeSequence [](start = 57, length = 19)

this behavior doesn't seem to belong to Utils. This is port assignment that belongs to scheduling behavior.
please refactor the code to move scheduling behavior out of Utils.java